### PR TITLE
rework firmware upgrade process

### PR DESCRIPTION
**Pull request by costanic**
firmware upgrades will be handled by snap-refresh rather than
side-loading snaps through Pelion.

### CODE REVIEW

**Review by mray190 — APPROVED** [review]

File: files/edge-core/scripts/edge-core-bootloader.sh [hunk 1]
```diff
@@ -70,53 +57,87 @@ WATCH_ID_STATUS_NONE=1
...
+    # remove the upgrade tgz file so that we don't fall into an upgrade loop
+    rm "${UPGRADE_TGZ}"
+    # TODO: init fota process state tracking
+fi
 
-check_snap_refresh
-# suggested usage: check_snap_refresh && copy_hdr_bin
+# move into folder and call pre-refresh if exists
+retval=0
+if [ -e "${UPGRADE_WORKDIR}"]; then
```
Comment (costanic) [inline files/edge-core/scripts/edge-core-bootloader.sh:98]: does this line contain a syntax error?  should there be a space after the closing quote and before the bracket?

**Review by johnmichaelwelsh — APPROVED** [review]
LGTM

File: files/edge-core/scripts/edge-core-bootloader.sh (hunk 1 above)
Comment (moonchen) [reply files/edge-core/scripts/edge-core-bootloader.sh]: Fixed

File: files/edge-core/scripts/edge-core-bootloader.sh [hunk 2]
```diff
@@ -73,7 +73,8 @@ function check_snap_refresh() {
 		log_msg "Waiting for snap refresh ID $watch_id, max $timeout seconds"
 		local end_msg="did not complete in time"
 		while [ $tdiff -lt $timeout ]; do
-			status=$(curl -sS --unix-socket /run/snapd.socket http://localhost/v2/changes/$watch_id | jq .result.status)
+			status=$(curl -sS --unix-socket /run/snapd.socket http://localhost/v2/changes/$watch_id | jq -r .result.status)
+            log_msg Snap refresh status: $status
```
Comment (costanic) [inline files/edge-core/scripts/edge-core-bootloader.sh:77]: this looks like a tabs vs spaces error here

**Review by kylestein-arm — APPROVED** [review]

**Review by johnmichaelwelsh — APPROVED** [review]
LGTM

File: files/edge-core/scripts/edge-core-bootloader.sh (hunk 2 above)
Comment (moonchen) [reply files/edge-core/scripts/edge-core-bootloader.sh]: Removed the line with spaces.

**Review by mray190 — APPROVED** [review]

**Pull request merged**